w277901423 Bug: Workers Unable to Pull in us-east-1

whatisdot opened the pull request:
Closes https://www.wrike.com/open.htm?id=277901423

## What
This enables us to pass the vpc_id for ECS services to use when updating. This is a short-term fix that unblocks deployments to N.Virginia (us-east-1). Long term fix will require a default VPC to be correctly configured in each region.

This also updates the CPU and MemoryReservation attributes to give containers more resources by default. This is a stop-gap solution until we make this more configurable.

## Why
Services were choosing the VPC we use for Active Directory, due to no default being available. This blocked requests from the ECS Service to other AWS services.

## How
Add a CLI parameter to `git_push_eb`.

## Testing
1. Checkout this git-scripts branch (`w277901423-bug-workers-unable-to-pull-in`)
1. Choose a repository that has workers as ECS Services.
1. Delete the service from ECS (it will be re-created on deploy).
1. Run `git_push_eb --vpc_id vpc-c078d9a9 <ENV NAME>`
1. Navigate to the Service in the ECS cluster for your environment.
1. Look under *Allowed subnets* and confirm that the following values are now under it:
  * subnet-69a21200
  * subnet-9516d3ee
  * subnet-705d443a
1. Delete the service again.
1. Run `git_push_eb <ENV NAME>`
1. Navigate to the ECS Service and confirm that the subnets have been changed to:
  * subnet-ef6c7f97
  * subnet-bcf633d5
  * subnet-343a0a7e

whatisdot commented on lib/aws/ecs.py:
```diff
@@ -13,10 +13,10 @@ def register_task_definition(cls, params):
+      'cpu': '1024',
       'executionRoleArn': 'arn:aws:iam::'+get_account_id()+':role/ecsTaskExecutionRole',
       'family': params['family'],
-      'memory': '512',
```
`cpu` and `memory` attributes have some conditional logic. See [AWS Task Definition parameter documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size) for more details.

jeet4320 commented on lib/push_eb.py:
```diff
@@ -110,6 +110,7 @@ def parse_args(self):
     parser.add_argument('-r', '--region', help='aws region')
     parser.add_argument('-v', '--verbose', action='store_true')
     parser.add_argument('-d', '--debug', action='store_true')
+    parser.add_argument('--vpc_id', nargs='?')
```
I understand name suggests vpc id, but I think some info under `help` section would help as it it not recommended to change vpcs often during deployment.

whatisdot commented:
@jeet4320 I've added a description. I also noticed that the environment wasn't being required as an argument, or set during execution so I added that as well.

jeet4320 commented on lib/aws/ecs.py:
```diff
@@ -13,10 +13,10 @@ def register_task_definition(cls, params):
       'executionRoleArn': 'arn:aws:iam::'+get_account_id()+':role/ecsTaskExecutionRole',
       'family': params['family'],
-      'memory': '512',
+      'memory': '3072',
```
are we using 3 GB memory in heroku for worker? I see for snaptx qa5 worker it is 512 mb

whatisdot commented:
These updates will conflict with existing environments. I'm putting Not Ready on this until I can make this backward compatible.

whatisdot commented:
We will create a new VPC, which will trivialize this. The memory issue can be addressed next week when we discuss the configuration file format. Closing this issue for now.
